hera_cal/frf.py

```diff
@@ -197,16 +194,15 @@ def timeavg_waterfall(data, Navg, flags=None, nsamples=None, rephase=False, lsts
     # wrap lsts
     avg_lsts = avg_lsts % (2 * np.pi)
 
-    return dict(avg_data=avg_data, win_flags=win_flags, avg_nsamples=avg_nsamples,
-                avg_lsts=avg_lsts, avg_extra_arrays=avg_extra_arrays)
+    return avg_data, win_flags, avg_nsamples, avg_lsts, avg_extra_arrays
 
 
 class FRFilter(VisClean):
```

Comment:
I have a more general question. What's the point of making FRF a subclass of any VisClean? Is it so that you can time axis cleaning? Maybe I'm confused because I don't know what a FRF run function will look like.

Reply:
Yeah, I guess the point is so that you can do some cleaning on the data before time averaging it, if you like, perhaps for example to fill-in entire integrations that were flagged, which would throw power for a true FR filter (rather than a FR clean as `VIsClean` is doing).

Reply:
I'm still not totally convinced, but I'm ok with you closing this thread if you address the one below: https://github.com/HERA-Team/hera_cal/pull/399/files#r247688297